src/backend/api/meals.js

```diff
@@ -1,15 +1,97 @@
+const { response, request } = require("express");
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
+
+    
+    if('maxPrice' in request.query){
+      const maxPrice = request.query.maxPrice;
+      const highPriceMeals = await knex ('meal').where ('price', '<=', maxPrice);
+      response.send(highPriceMeals);
+      return;
+    }
+    if('availableReservations' in request.query){
+      const availableReservations = (request.query.availableReservations == 'true');
+      const numberOfReservations = await knex.from('meal').join('reservations','meal.id', '=', 'reservations.id')
+      .select('meal.title')
+      .where(('max_reservations' - 'reservations.number_of_guests', true), availableReservations)
+      response.send(numberOfReservations);
+      return;
+    }
+    if('title' in request.query){
+      const title = request.query.title;
+      const mealsTitle = await knex('meal').where('title', 'like', '%' + title + '%');
+      response.send(mealsTitle);
+      return;
+    }
+    if('createdAfter' in request.query){
+      const createdAfter = request.query.createdAfter;
+      const dateCreatedAfter = await knex('meal').where('created_date', '>=',createdAfter);
+      response.send(dateCreatedAfter);
+    }
+    if('limit' in request.query){
+       const limit = request.query.limit;
+       const mealsLimit = await knex('meal').limit(limit);
+       response.send(mealsLimit);
+       return;
+    }
+  
+  const meal = await knex("meal");
+    response.send(meal);
+  } catch (error) {
+    throw error;
+  }
+});
+    
+ /*  // knex syntax for selecting things. Look up the documentation for knex for further info
     const titles = await knex("meals").select("title");
     response.json(titles);
   } catch (error) {
     throw error;
   }
+});*/
+
+router.get('/:id', async (req,res) =>{
+  try {
+    const mealId = req.params.id;
+    const requestedMeal = await knex('meal').where ('id',mealId);
+    res.send(requestedMeal[0]);
```

Comment:
```suggestion
    res.send(requestedMeal);
```
the `id` should be unique...therefore you can get only one value or error